distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -1057,6 +1045,24 @@ type Builder
             self.unsafe_append item
             self
 
+    ## Appends a part of a given vector to this builder
+
+       Arguments:
+       - vector: The vector from which the elements are sourced.
+       - start: The start index of the range to append.
+       - end: The end index (the first index after the last element to be
+         appended) of the range to be appended.
+
+       > Example
+         Append a part of the vector.
+
+             builder = Vector.new_builder
+             builder . append_vector_range [20, 30, 40, 50] 1 3 . to_vector == [30, 40]
+    append_vector_range : Vector Any ! Error -> Builder ! Error
+    append_vector_range self vector start end =
```

Comment:
Shouldn't this method also use the `Index_Sub_Range` parameter like `take` and `drop`?

Reply:
Well, I'd say not - this is a lower-level helper - actually we then _use_ `append_vector_range` when implementing `take` and `drop` - so we'd be a bit circular.

Alternatively, we can rename this method to something like `internal_append_vector_range` and then provide one like you suggest. However I think for now I'd keep it as-is and revisit it when it is ever needed - currently I'd say that the `Index_Sub_Range` is mostly aimed at regular GUI users whereas the `Vector.Builder` is something reserved more for power-users and I don't think GUI should use it often, if at all.